tests/lib/hardware/hardware_base.py

```diff
@@ -79,6 +80,17 @@ def generate_keys(self):
             "%s%s_key" % (config.CLUSTER_PREFIX, self.hardware_uuid)
         self.pubkey = "%s %s" % (key.get_name(), key.get_base64())
 
+    def ssh_agent(self):
+        res = subprocess.run(['ssh-agent'], capture_output=True)
```

Comment:
please use `check_call` or something that does check the return code.

Reply:
Actually since python 3.5 the doc recommand run() method for most everything.
https://docs.python.org/3/library/subprocess.html

Reply:
then `check=True` in `run()`

Reply:
Forgot that.
Done now.